src/Middleware/Diagnostics/src/DiagnosticsTelemetry.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Diagnostics;
+
+internal static class DiagnosticsTelemetry
+{
+    public static void ReportUnhandledException(ILogger logger, HttpContext context, Exception ex)
```

Comment:
Would an extension method have a better change of getting called over the default `UnhandledException`?

Reply:
I don't think an extension method makes sense here. Realistically this will only ever be called from two places. I want DRY to avoid the two middleware from going out of sync.